packages/core/src/internal/base/button.base.ts

```diff
@@ -73,6 +75,11 @@ export class CdsBaseButton extends LitElement {
     `;
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+    this.tabIndex = 0; // initialize immediately so button can be focused synchronously
```

Comment:
This allows users to reference and call `focus()` on an element immediately. Later in the lifecycle if the button is set to readonly the `tabindex` is removed.